torch/fx/experimental/GraphManipulation.py

```diff
@@ -121,6 +121,23 @@ def serialize_weight(tensor: torch.Tensor) -> Dict:
     return weight
 
 
+def serialize_leaf_module(
+    mod: torch.nn.Module, weights_metadata: Dict, weights: Dict, name_prefix: str
+) -> Dict:
+    parameters: Dict[str, Any] = {}
+    parameters["name"] = type(mod).__name__
+    for name, buffer in mod.named_buffers():
+        weights_metadata[f"{name_prefix}.{name}"] = serialize_weight(buffer)
+        weights[f"{name_prefix}.{name}"] = buffer
+    for name, parameter in mod.named_parameters():
+        weights_metadata[f"{name_prefix}.{name}"] = serialize_weight(parameter)
+        weights[f"{name_prefix}.{name}"] = parameter
+    if isinstance(mod.__constants__, List):
```

Comment:
OK, I understand what's going on now. `__constants__` is here because the standard `nn.Module` instances just happen to have those defined for TorchScript compatibility. This is *not* a part of the `nn.Module` API, thus this will break in the general case. My suggestion is to write functions for each `nn.Module` type to pull out the pertinent attributes for lowering

Reply:
And for more constants, these field names are listed in `__constants__` because they're used in `forward()`, there is no contract that says that `__constants__` contains all of the pertinent information for lowering this `Module`

Reply:
Good catch, per our offline discussion I'll go ahead and land this for now to unblock testing but @842974287 will follow up with a more complete paramater harvesting pass and we'll update the JSON serialization then.